### PR TITLE
Iterbale refactor

### PR DESCRIPTION
**Pull request by ruchiramoitra**

### CODE REVIEW

File: __tests__/data/iterable_input.json [hunk 1]
```diff
@@ -908,5 +908,58 @@
       },
       "Enabled": true
     }
+  },
+  {
```
Comment (arnab-p) [inline __tests__/data/iterable_input.json:912]: This test is not passing

File: v1/destinations/iterable/data/IterableIdentifyConfig.json [hunk 2]
```diff
@@ -2,14 +2,16 @@
   {
     "destKey": "email",
     "sourceKeys": [
-      "context.traits.email"
+      "context.traits.email",
```
Comment (arnab-p) [inline v1/destinations/iterable/data/IterableIdentifyConfig.json:5]: Need to put top-level traits with higher precedence. The same comment goes for similar changes.

File: __tests__/data/iterable_input.json (hunk 1 above)
Comment (ruchiramoitra) [reply __tests__/data/iterable_input.json:912]: done

File: v1/destinations/iterable/data/IterableIdentifyConfig.json (hunk 2 above)
Comment (ruchiramoitra) [reply v1/destinations/iterable/data/IterableIdentifyConfig.json]: changed

**Pull request merged**